ksql-engine/src/test/java/io/confluent/ksql/EndToEndEngineTestUtil.java

```diff
@@ -70,6 +74,43 @@
     UdfLoaderUtil.load(new MetaStoreImpl(functionRegistry));
   }
 
+  private static class ValueSpec {
+    private final Object spec;
+
+    ValueSpec(final Object spec) {
+      this.spec = spec;
+    }
+
+    private static void compare(Object o1, Object o2, String path) {
```

Comment:
Should we make the `path` `Optional`? i.e, i see that from equals it is called with `""` which would result in paths like `.blah.foo`

Reply:
I'll make that change. I wasn't too bothered by it at first since this string is just for our own consumption.